trestle/core/markdown/md_writer.py

```diff
@@ -162,3 +168,17 @@ def get_lines(self) -> List[str]:
     def get_text(self) -> str:
         """Get the text as currently written."""
         return '\n'.join(self._lines)
+
+    def _add_subnode_text(self, node: MarkdownNode) -> None:
+        """Add the header text to the markdown contents."""
+        for line in node.content.raw_text.split('\n'):
+            self.new_line(line)
+        for subnode in node.subnodes:
+            self._add_subnode_text(subnode)
+
+    def write_out_header_tree(self, header: Dict[str, Any], tree: MarkdownNode) -> None:
+        """Write out the header and markdown node as markdown file."""
+        self.add_yaml_header(header)
+        for node in tree.subnodes:
+            self._add_subnode_text(node)
```

Comment:
tree.content.raw_text contains all markdown, no need to traverse children

Reply:
reverted the change